morpho-checkpoint-node/src/Morpho/Tracing/Pretty.hs

```diff
@@ -88,13 +138,66 @@ instance
       MorphoStdHeader {..} = morphoHeaderStd
 
 instance MPretty MorphoBlockTx where
-  mpretty = mpretty . morphoBlockGenTx
+  mpretty (MorphoBlockTx genTx genTxId) =
+    "MorphoBlockTx(transaction:"
+      <+> mpretty genTx
+      <> ", transactionId:"
+      <+> mpretty genTxId
+      <> ")"
+
+instance MPretty (ChainHash (MorphoBlock h c)) where
+  mpretty GenesisHash = "GenesisHash"
+  mpretty (BlockHash h) = "BlockHash(" <> mpretty h <> ")"
+
+instance MPretty (HeaderHash a) => MPretty (Tip a) where
+  mpretty TipGenesis = "TipGenesis"
+  mpretty (Tip s h b) =
+    "Tip(slot:"
+      <+> mpretty s
+      <> ", hash:"
+      <+> mpretty h
+      <> ", blockNo:"
+      <+> mpretty b
+      <> ")"
 
 instance MPretty Tx where
   mpretty (Tx vote) = mpretty vote
 
 instance MPretty Vote where
-  mpretty = viaShow
+  mpretty (Vote block _) = mpretty block
```

Comment:
We can also use `recoverPublicKey` here to know which node signed it. I'd go for a format like `Vote(block: \<block\>, byPubkey: \<pubkey\>)`